src/shouldHideEvent.js

```diff
@@ -27,10 +27,11 @@ function memberEventDiff(ev) {
     const content = ev.getContent();
     const prevContent = ev.getPrevContent();
 
-    diff.isJoin = content.membership === 'join' && prevContent.membership !== 'ban';
-    diff.isPart = content.membership === 'leave' && ev.getStateKey() === ev.getSender();
-
     const isJoinToJoin = content.membership === prevContent.membership && content.membership === 'join';
+
+    diff.isJoin = !isJoinToJoin && content.membership === 'join' && prevContent.membership !== 'ban';
+    diff.isPart = !isJoinToJoin && content.membership === 'leave' && ev.getStateKey() === ev.getSender();
```

Comment:
But a "part" could never be a join to join.. so `!isJoinToJoin` is always true here.

Reply:
```
const isMembershipChanged = content.membership !== prevContent.membership;
diff.isJoin = isMembershipChanged && ...;
diff.isPart = isMembershipChanged && ...;
```

is what I had before @t3chguy beat me to the PR.